onnxruntime/core/providers/cpu/nn/pool.cc

```diff
@@ -216,9 +218,15 @@ Status Pool<float, AveragePool>::Compute(OpKernelContext* context) const {
 
 template <>
 Status Pool<float, MaxPool<8 /*VERSION*/>>::Compute(OpKernelContext* context) const {
-  // Use MLAS pooling if the index output tensor is not used.
-  if (OpKernel::Node().OutputDefs().size() == 1) {
-    return PoolBase::Compute(context, MlasMaximumPooling);
+  // Use MLAS pooling if the index output tensor is not used
+  // and also if dilation is not required
+
+  bool need_dilation = false;
```

Comment:
i guess this variable is not needed any where?

Reply:
It's used in the code block below. We can use a more optimized algorithm if dilations is not required.

   if (OpKernel::Node().OutputDefs().size() == 1 && !need_dilation)